app/lib/features/chat_ng/widgets/events/text_message_event.dart

```diff
@@ -1,153 +1,112 @@
-
-class TextMessageEvent extends ConsumerWidget {
+class TextMessageEvent extends ConsumerStatefulWidget {
   final String roomId;
   final MsgContent content;
-  final TextMessageType _type;
-  final bool _isMe; // Only needed for emoji messages to determine style
+  final bool isNotice;
+  final bool isReply;
   final Widget? repliedTo;
 
-  const TextMessageEvent.inner({
+  const TextMessageEvent({
     super.key,
     required this.content,
     required this.roomId,
-    required TextMessageType type,
-    bool isMe = false,
+    this.isNotice = false,
+    this.isReply = false,
     this.repliedTo,
-  }) : _type = type,
-       _isMe = isMe;
+  });
 
-  factory TextMessageEvent.emoji({
-    Key? key,
-    required MsgContent content,
-    required String roomId,
-    required bool isMe,
-  }) {
-    return TextMessageEvent.inner(
-      key: key,
-      content: content,
-      roomId: roomId,
-      type: TextMessageType.emoji,
-      isMe: isMe,
-    );
-  }
+  @override
+  ConsumerState<TextMessageEvent> createState() => _TextMessageEventState();
+}
 
-  // Factory constructor for reply messages
-  factory TextMessageEvent.reply({
-    Key? key,
-    required MsgContent content,
-    required String roomId,
-  }) {
-    return TextMessageEvent.inner(
-      key: key,
-      content: content,
-      roomId: roomId,
-      type: TextMessageType.reply,
-    );
-  }
+class _TextMessageEventState extends ConsumerState<TextMessageEvent> {
+  late String body;
+  String? bodyFormatted;
 
-  // Factory constructor for notice messages
-  factory TextMessageEvent.notice({
-    Key? key,
-    required MsgContent content,
-    required String roomId,
-    Widget? repliedTo,
-  }) {
-    return TextMessageEvent.inner(
-      key: key,
-      content: content,
-      roomId: roomId,
-      type: TextMessageType.notice,
-      repliedTo: repliedTo,
-    );
+  @override
+  void initState() {
+    super.initState();
+    bodyFormatted = widget.content.formattedBody();
+    body = widget.content.body();
   }
-  // Default factory constructor
-  factory TextMessageEvent({
-    Key? key,
-    required MsgContent content,
-    required String roomId,
-    Widget? repliedTo,
-  }) {
-    return TextMessageEvent.inner(
-      key: key,
-      content: content,
-      roomId: roomId,
-      type: TextMessageType.regular,
-      repliedTo: repliedTo,
-    );
+
+  @override
+  void didUpdateWidget(TextMessageEvent oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    bodyFormatted = widget.content.formattedBody();
+    body = widget.content.body();
   }
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final textTheme = Theme.of(context).textTheme;
-    final chatTheme = Theme.of(context).chatTheme;
-    final colorScheme = Theme.of(context).colorScheme;
-    final body = content.formattedBody() ?? md.markdownToHtml(content.body());
+  Widget build(BuildContext context) {
+    final inner =
+        (isOnlyEmojis(body))
```

Comment:
maybe we actually want to store that flag, too, but the regexp isn't too expensive yet...